docs/_posts/2015-12-09-ismounted-antipattern.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "isMounted is an Antipattern"
+author: jimfb
+---
+
+The primary use case for `isMounted()` is to avoid calling `setState()` after a component has unmounted, because calling `setState()` after a component has unmounted will emit a warning. The “setState warning” exists to help you catch bugs, because calling `setState()` on an unmounted component is an indication that your app/component has somehow failed to clean up properly. Specifically, calling `setState()` in an unmounted component means that your app is still holding a reference to the component after the component has been unmounted - which often indicates a memory leak!
+
+To avoid the error message, people often add lines like this:
+
+```js
+if(this.isMounted()) this.setState({...}); // This is bad.
+```
+
+Checking `isMounted` before calling `setState()` does eliminate the warning, but it also defeats the purpose of the warning, since now you will never get the warning (even when you should!)
+
+Most other uses of `isMounted()` are similarly erroneous; using `isMounted()` is usually a code smell.
+
+An easy migration strategy for anyone upgrading their code to avoid `isMounted()` is to track the mounted status yourself.  Just set a `_isMounted` property to true in `componentDidMount` and set it to false in `componentWillUnmount`, and use this variable to check your component's status.
+
+An optimal solution would be to find places where `setState()` might be called after a component has unmounted, and fix them. Such situations most commonly occur due to callbacks, when a component is waiting for some data and gets unmounted before the data arrives. Ideally, any callbacks should be canceled in `componentWillUnmount`, prior to unmounting.
+
+If you use ES6 promises, you may need to wrap your promise in order to make it cancelable.
+
+```js
+const cancelablePromise = makeCancelable(new Promise(r => setTimeout(r, 1000)));
+
+cancelablePromise
+  .promise
+  .then(() => console.log('resolved'))
+  .catch((reason) => console.log('isCanceled', reason.isCanceled));
+
+cancelablePromise.cancel(); // Cancel the promise
+```
+
+Where `makeCancelable` is defined by [@istarkov](https://github.com/istarkov) as:
```

Comment:
It was mentioned deep in the middle of a long github issue (https://github.com/facebook/react/issues/5465#issuecomment-157888325).  I didn't really want to link there, as it would confuse people.